### PR TITLE
feat: better readme

### PR DESCRIPTION
**Pull request by kilted-andres**
## fixes KILTProtocol/ticket#2828

A new extended `Readme`-file that should help doing the documentation. (Or even replace it). 

I think that it is better if someone, who has not used this repository judges how good the explanations are. 
Please, clone the repo and try to use it before any review. 
Please, tell me what is unclear and what else should be inside.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,45 @@
...
 
-2. Domain linkage credential
 
-- Bind your DID to a specific domain
-- This prevents Man-in-the-Middle attacks
+If you don't have some of the above, don't worry, we help you get them on the sections below. 
 
-3. A CType you want to request from the user
+After setting up and running this project locally on your computer, you will know how the [login process](#Process) works and have code that you can implement on your website. 
+We encourage you to customize the code for your specific usecase, just keep in mind the [specifications](https://github.com/KILTprotocol/spec-ext-credential-api) to retain compatibility.
```
Comment (Ad96el) [inline README.md:38]: Is this a boilerplate or tutorial?

File: README.md [hunk 2]
```diff
@@ -1,29 +1,45 @@
...
-- This prevents Man-in-the-Middle attacks
+If you don't have some of the above, don't worry, we help you get them on the sections below. 
 
-3. A CType you want to request from the user
+After setting up and running this project locally on your computer, you will know how the [login process](#Process) works and have code that you can implement on your website. 
+We encourage you to customize the code for your specific usecase, just keep in mind the [specifications](https://github.com/KILTprotocol/spec-ext-credential-api) to retain compatibility. 
 
-- In this example, we use the email CType, but you could use any CType you want
+If you want to implement KILT Login without understanding it, we also have a solution for you. 
+There is a [containerized version under this repository](https://github.com/KILTprotocol/kilt-login).
```
Comment (Ad96el) [inline README.md:41]:
```suggestion
As mentioned, this is an educational project and is not suitable for production. There is a production-ready solution that can be easily integrated into your infrastructure.
```

File: README.md [hunk 3]
```diff
@@ -46,26 +67,61 @@ Defining them is part of the set up of your project.
...
 - `DAPP_NAME` = _This should be a custom name for your dApp_
 - `JWT_SIGNER_SECRET` = _This is secret key (string) that signs the Json-Web-Tokens before saving them in the Cookies_
 
-There is a script to facilitate the generation of the environment variables that are needed to set up your decentralized App (dApp).
-This script is called `./scripts/genesisEnvironmentVariables.ts`.
-You can execute it by running `yarn environment`.
+### How do I get mine? 
+You have three different options: 
+
+1. Using `yarn environment`:
```
Comment (Ad96el) [inline README.md:86]: Did not worked. I copied the env example renamed and executed yarn environment

File: README.md [hunk 4]
```diff
@@ -46,26 +67,61 @@ Defining them is part of the set up of your project.
...
+
+After having all your enviorement variables, to build a **Domain Linkage Credential**:
+
++ run: `yarn well-known-config`.
+
+If you want know more about this, check out the [Identity Foundation Documentation](https://identity.foundation/.well-known/resources/did-configuration/). 
+[Our dApp documentation also have a section about this.](https://docs.kilt.io/docs/develop/dApp/well-known-did-config)
+
+## Build 
+After having all your enviorement variables and your well-known-did-configuration:
```
Comment (Ad96el) [inline README.md:117]: why is this in the documentation?

**Review by Ad96el — COMMENTED** [review]
I am struggling to understand what the purpose of the README is. 

It is neither a tutorial How-To guide reference nor explanation. 

I would argue that the README should be just a reference where the user can quickly understand the system components and provide instructions how to quickly run the system -not more. The README should point to the tutorial.

File: README.md (hunk 1 above)
Comment (kilted-andres) [reply README.md:38]: Well, both. But more of a tutorial.

File: README.md [hunk 5]
```diff
@@ -33,11 +49,16 @@ If you are not familiar with some of these, we recommend to first get an overvie
 - [decentralized identifier (DID)](https://docs.kilt.io/docs/concepts/did)
 - [typescript](https://www.typescriptlang.org/)
 - [JSON](https://www.json.org/json-en.html)
-- [JSON Web Token](https://en.wikipedia.org/wiki/JSON_Web_Token)
+- [JSON Web Token (JWT)](https://en.wikipedia.org/wiki/JSON_Web_Token)
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:52]: Do you mean this: https://datatracker.ietf.org/doc/html/rfc7519?

File: .env.example [hunk 6]
```diff
@@ -0,0 +1,8 @@
+WSS_ADDRESS=wss://peregrine.kilt.io/
+FRONTEND_PORT=8080
+BACKEND_PORT=3000
+DAPP_ACCOUNT_MNEMONIC=coin coin coin coin coin coin coin coin coin coin coin coin
+DAPP_DID_MNEMONIC=identify identify identify identify identify identify identify identify identify identify identify identify
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply .env.example:5]: Yes, they have to create one. 

I just wanted to show the format. Here 12 words from the BIP.

File: README.md [hunk 7]
```diff
@@ -1,29 +1,45 @@
 # web3-login-demo
 
-This website demonstrates how to build a logging using KILT Protocol.
+This project demonstrates how to enable **login using KILT Protocol**. 
+It is an educational project, everything is broken down to little steps and has explanations to ease your understanding.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:4]: I could add a "Steps" section that list the order in wicht to follow the sections. But they are already in that order.

File: README.md [hunk 8]
```diff
@@ -1,29 +1,45 @@
...
+During the login procedure, the user is required to present a credential.
 You can decide which credentials to accept.
 They could be issued by yourself or by other attesters.
-In this demo we request a credential that contains an email address that the user owns.
-For that we rely on https://socialkyc.io to issue email credentials.
+
+This is basically like asking for an ID-Card instead of a customer number, in order to provide a personalized service.
+This means that users don't not need to setup an account and password just for your website.
+Additionally, this avoids third parties (usually powerfull data-collecting companies) from tracking your interactions with your clients.
+Not even the KILT developers can track it.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:13]: How would you describe that. I try to do that with line 11.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: I updated it to that.

File: README.md [hunk 9]
```diff
@@ -1,29 +1,45 @@
...
+- Bind your DID to a specific domain.
+- This prevents Man-in-the-Middle attacks.
+- Also known as the _well-known-did-configuration_.
 
-- This DID is used so that the user knows the parties to whom they talk
+3. A CType to request from the user
+- The type of credentials the dApp considers valid. 
+- In this demo, as a default, we request a credential that contains an email address.
+For that we rely on [SocialKYC](https://socialkyc.io) to issue email credentials after verifing the user owns it.
+You could easily modified this.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:32]: Maybe I should add a section about this on the bottom.

File: .env.example [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+DAPP_DID_URI=did:kilt:4xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply .env.example]: It's all empty now.

[the remark this answers is not in the document]
Comment (kilted-andres) [reply .env.example]: All variables are empty now.

File: README.md (hunk 2 above)
Comment (kilted-andres) [reply README.md:41]: This is also suitable for production. There are Dockerfiles and everything. There is a deployed version. 

The difference with the containerised version is that you would actually integrated with your website instead of having a middle man doing the work for you. 

Your website would actually work with KILT Credentials instead of just a JWT.

File: README.md (hunk 4 above)
Comment (kilted-andres) [reply README.md:117]: The link to our documentation on the same topic?

File: README.md (hunk 3 above)
Comment (kilted-andres) [reply README.md:86]: Good catch. Thanks!

Empty declarations like `ENV_NAME=""` where leading to jump the step.

File: README.md (hunk 1 above)
Comment (kilted-andres) [reply README.md:38]: It is a tutorial, but you can use the code. Like in any other tutorial, isn't it?

File: README.md [hunk 11]
```diff
@@ -46,27 +77,78 @@ Defining them is part of the set up of your project.
...
+- run `yarn start`.
+
+## Build
+
+To get a production version of the app, you can bundle the frontend of the app into static files.
+For this, after having all your environment variables and your well-known-did-configuration:
+
+- run `yarn build`.
+
+## Login Process
```
Comment (Ad96el) [inline README.md:141]: From my point of view, this should not be part of this README. This is explanation and should be in a tutorial.

File: README.md [hunk 12]
```diff
@@ -1,29 +1,54 @@
...
+It is an educational project, everything is broken down to little steps and has explanations to ease your understanding.
+
+During [the login procedure](#login-process), the user is required to present a credential.
 You can decide which credentials to accept.
 They could be issued by yourself or by other attesters.
-In this demo we request a credential that contains an email address that the user owns.
-For that we rely on https://socialkyc.io to issue email credentials.
+
+This is basically like asking for an ID-Card instead of a customer number, in order to provide a personalized service.
+This means that users don't not need to setup an account and password just for your website.
```
Comment (Ad96el) [inline README.md:15]: don't not

**Review by Ad96el — COMMENTED** [review]
I think it is much better now. There are still explanations that I believe should not be part of the README. Perhaps I am mistaken, but I would prefer to see the login process and much of the introductory content in a tutorial instead."

File: README.md (hunk 12 above)
Comment (kilted-andres) [reply README.md:15]: Double negation.

 So... "yes, do"?

File: README.md (hunk 11 above)
Comment (kilted-andres) [reply README.md:141]: The `login process` or the `build`?

File: README.md (hunk 2 above)
Comment (kilted-andres) [reply README.md]: Maybe you like it better now.

File: README.md [hunk 13]
```diff
@@ -46,27 +114,71 @@ Defining them is part of the set up of your project.
...
+
+After having all your environment variables and your well-known-did-configuration:
+
+- run `yarn start`.
+
+## Login Process
+
+After having set up the whole project, when the website is up and running, the login process can take place.
+Each user that wants to login would trigger the process that is displayed below.
+The process is full of HTTPS queries and extension-api's messages.
```
Comment (weichweich) [inline README.md:175]: If you run it locally it's not HTTPS. What should this sentence tell the developer? It currently conveys that the process contains many HTTPS queries with the connotation that it's very complicated.

I would rather avoid that.

File: README.md [hunk 14]
```diff
@@ -46,27 +114,71 @@ Defining them is part of the set up of your project.
...
+
+A user could trigger the whole process with just one click on the website, for example "Login with extension X" or "Login with extension Y".
+
+On this example project, several (2-3) interactions are need because it is broken down into little steps.
+We encourage you to open and read the browser's and backend's consoles.
+Try to understand what is happening on each step, which function is responsible for what, how did the cookies changed.
```
Comment (weichweich) [inline README.md:181]: This sounds like it's a Uni assignment. 🎓 Or maybe a challenge. They shouldn't `try to understand`. They should simply read and understand

**Review by weichweich — CHANGES_REQUESTED** [review]
Nice tutorial! I'd like some changes though

File: .env.example [hunk 15]
```diff
@@ -0,0 +1,8 @@
+WSS_ADDRESS=wss://peregrine.kilt.io/
+FRONTEND_PORT=6565
+BACKEND_PORT=2525
+DAPP_ACCOUNT_MNEMONIC=
+DAPP_DID_MNEMONIC=
+DAPP_DID_URI=
+DAPP_NAME=
+JWT_SIGNER_SECRET=
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply .env.example]: `eslint` ignores the shell files.

I added the line now. Couldn't found in internet how to add this as a `eslint` rule that applies to `.env.example`.

File: README.md [hunk 16]
```diff
@@ -1,29 +1,91 @@
...
+This project demonstrates how to enable **login using KILT Protocol**.
+It is an educational project, everything is broken down to little steps and has explanations to ease your understanding.
+
+During [the login procedure](#login-process), the user is required to present a credential.
 You can decide which credentials to accept.
 They could be issued by yourself or by other attesters.
-In this demo we request a credential that contains an email address that the user owns.
-For that we rely on https://socialkyc.io to issue email credentials.
+
+This is basically like asking for an ID-Card instead of a customer number, in order to provide a personalized service.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:14]: Ok. Yes, I guess some people run away from the idea of showing their government ID-Card.

An Identification card could be anything though; like from a university, a club, a company, etc. Not necessarily a membership.  But is true that people first think of government credentials. Better to avoid it.



I will replace line 14 with your approach.

[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: done

File: README.md [hunk 17]
```diff
@@ -1,29 +1,91 @@
...
+This means that users don't not need to setup an account and password just for your website.
+Additionally, this avoids third parties (usually powerful data-collecting companies) from tracking your interactions with your clients.
+Not even the KILT developers can track it.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:19]: It only avoids tracking from third parties; like Google or Facebook or Apple. 

But the Service provider, the website you are logging in, can definitely track you. 

So it is only less tracking compared to "authorization servers" issuing access tokens. So, implementations of the [Open Authorization Protocol](https://oauth.net/2/).

File: README.md [hunk 18]
```diff
@@ -1,29 +1,91 @@
...
+- Bind your DID to a specific domain.
+- This prevents Man-in-the-Middle attacks.
+- Also known as the _well-known-did-configuration_.
+
+3. A CType to request from the user
+
+- The type of credentials the dApp considers valid.
+- The issuer of this credentials can also be important.
+  Your dApp can opt to accept only credentials coming from _trusted attesters_.
+  We recommend always doing this and it is default on this demo.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:38]: I'm against of separating the cType from the trusted attesters. Attesters are only trusted with issuing specific types of credentials. 
If they have independent lists of trusted attesters and of cTypes it wold be hard to know who is trusted with what. I wouldn't accept a passport issued by SocialKYC. 

It is true that is best to only accept credentials from trusted attesters, but checking the attester is not part of the SDK. ` Kilt.Credential.verifyPresentation(params)` does not check that. That is way I say it is optional. 

I will try to rephrased it on way that makes it clear that not checking for the issuer is not secure.

File: README.md [hunk 19]
```diff
@@ -1,29 +1,91 @@
...
+
+If you don't have some of the above, don't worry, we help you get them on the sections below.
+
+After setting up and running this project locally on your computer, you will know how the [login process](#Process) works.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:43]: I just wanted to mention the scope and objective of this project. 

Maybe "you will know" is too much. Gonna try to rephrase it.

File: README.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: Hope it is better now.

File: README.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:43]: Is this better?  

```
After setting up and running this project locally on your computer, you will have representation of how the [login process](#Process) works.
```

File: README.md [hunk 20]
```diff
@@ -1,29 +1,91 @@
...
+You are free to copy code from here, but we do encourage you to customize the code for your specific use case.
+Just keep in mind the [specifications](https://github.com/KILTprotocol/spec-ext-credential-api) to retain compatibility with the wallets/extensions.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:45]: Followed it!

File: README.md [hunk 21]
```diff
@@ -1,29 +1,91 @@
...
+
+If you want to implement KILT Login without understanding it, we also have a solution for you.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:47]: It does suggest that, but it also suggest that it is ok. 

Everybody uses a mobile telephone and almost nobody knows how it works.

[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:47]: I'm not attached to that sentence, so I can change it to yours.

File: README.md [hunk 22]
```diff
@@ -1,29 +1,91 @@
...
+There is a [containerized version called openDID](https://github.com/KILTprotocol/openDID).
+On that solution the whole process takes place in a extra container and only a autehtification token is passed to the main website.
+
+## Users requirements
+
+It is off course important, that the dApp an the user interact with information from the same chain.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:53]: "Off course" never really adds anything. 

I can remove it, if it bothers you.

[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: gone

File: README.md [hunk 23]
```diff
@@ -1,29 +1,91 @@
...
+There is the **KILT Test Chain** aliased **_Peregrine_**.
+And there is the **real KILT Chain** aliased **_Spiritnet_**.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: It's production now.

File: README.md [hunk 24]
```diff
@@ -1,29 +1,91 @@
...
 
-- This DID is used so that the user knows the parties to whom they talk
+For a user to login using KILT, it needs:
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: changed to 

```
For a user to log in with KILT, it requires:
```

File: README.md [hunk 25]
```diff
@@ -33,11 +95,17 @@ If you are not familiar with some of these, we recommend to first get an overvie
...
+- [JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519)
 - [express.js](https://expressjs.com/)
 - [react.js](https://react.dev/)
 
-## Setup
+## Installation
+
+After [cloning the repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) to install all required modules,
+
+- run: `yarn install`.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:107]: The coma was on the wrong spot. 
It should be: "After cloning the repository, to install all required modules run: "

File: README.md [hunk 26]
```diff
@@ -1,29 +1,91 @@
...
 
-2. Domain linkage credential
+1. A compatible wallet or extension.
 
-- Bind your DID to a specific domain
-- This prevents Man-in-the-Middle attacks
+- This means they follow the [KILT credential api specifications](https://github.com/KILTprotocol/spec-ext-credential-api).
 
-3. A CType you want to request from the user
+- Get the **Sporran Wallet**
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: changed.

File: README.md [hunk 27]
```diff
@@ -1,29 +1,91 @@
...
 
-- In this example, we use the email CType, but you could use any CType you want
+  - for **_Peregrine_** [here](https://github.com/BTE-Trusted-Entity/sporran-extension/releases).
+  - for **_Spiritnet_** [here](https://www.kilt.io/services/sporran).
 
-If you don't have some of the above, make sure to go though the [setup section](#setup).
+2. An on-chain DID
+
+- If you are using Sporran, here is [how to get one.](https://www.kilt.io/build-your-identity)
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: changed it to "a DID".

File: README.md [hunk 28]
```diff
@@ -1,29 +1,91 @@
...
-- Bind your DID to a specific domain
-- This prevents Man-in-the-Middle attacks
+- This means they follow the [KILT credential api specifications](https://github.com/KILTprotocol/spec-ext-credential-api).
 
-3. A CType you want to request from the user
+- Get the **Sporran Wallet**
 
-- In this example, we use the email CType, but you could use any CType you want
+  - for **_Peregrine_** [here](https://github.com/BTE-Trusted-Entity/sporran-extension/releases).
+  - for **_Spiritnet_** [here](https://www.kilt.io/services/sporran).
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: the explanations for Peregrine and Spiritnet are right above this. I think the parenthesis are unnecessary.

[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: But the hyperlink describes itself better now. Good point, thanks!

File: README.md [hunk 29]
```diff
@@ -33,11 +95,17 @@ If you are not familiar with some of these, we recommend to first get an overvie
 - [decentralized identifier (DID)](https://docs.kilt.io/docs/concepts/did)
 - [typescript](https://www.typescriptlang.org/)
 - [JSON](https://www.json.org/json-en.html)
-- [JSON Web Token](https://en.wikipedia.org/wiki/JSON_Web_Token)
+- [JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519)
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:100]: @Ad96el is skeptical of wikipedia. RFC is more trustworthy. Thats why I changed it. 

I also think that wikipedia is a better starting point.

File: README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:19]: It's back to wiki.

File: README.md (hunk 29 above)
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: It's back to wiki.

File: README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (weichweich) [reply README.md:19]: `Not even the KILT developers can track it.` This sentence sounds like nobody could track it.

File: README.md (hunk 22 above)
[the remark this answers is not in the document]
Comment (weichweich) [reply README.md]: The reason I would like it removed is, that it actually adds a connotation. It suggests that this is obvious and should be known. We should not suggest such things in a guide. Thanks for removing

File: README.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (weichweich) [reply README.md]: Yeah sentences with comas are hard 😂. Best to avoid them. it's better now, but would be best with two sentences i think.

File: README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: changed to "There is no trace on the KILT blockchain about the interaction between you and your client."

File: README.md [hunk 30]
```diff
@@ -46,27 +114,71 @@ Defining them is part of the set up of your project.
...
 - `DAPP_ACCOUNT_MNEMONIC` = _This is the mnemonic of the Kilt account paying for all transactions_
 - `DAPP_DID_MNEMONIC` = _This is the mnemonic of the Kilt DID that identifies your dApp_
 - `DAPP_DID_URI` = _This is the URI of the Kilt DID that identifies your dApp_
 - `DAPP_NAME` = _This should be a custom name for your dApp_
 - `JWT_SIGNER_SECRET` = _This is secret key (string) that signs the Json-Web-Tokens before saving them in the Cookies_
 
-There is a script to facilitate the generation of the environment variables that are needed to set up your decentralized App (dApp).
-This script is called `./scripts/genesisEnvironmentVariables.ts`.
-You can execute it by running `yarn environment`.
+### How do I get mine?
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:133]: It is a subsection of `## Environment Variables`. I think it clear what are you getting.

[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:133]: Changed to `### Setup your environment` for the common happiness

File: README.md [hunk 31]
```diff
@@ -46,27 +114,71 @@ Defining them is part of the set up of your project.
...
+
+   This is a _Command Line Interface_ tool that can help you obtaining this variables and also does other common tasks (unrelated to this project).
+   The distillery uses the same key derivation as this repository, which means that it is highly compatible.
+
+3. Without help:
+
+   If you are a _pro_, you could defined and generate them externally and add them to the `.env`-file.
+   You would probably have to modify the `generateKeyPairs.ts` files (on _scripts_ and _backend_) to match your key derivation though.
+
+## Well-Known-DID-Configuration
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: Dehyphenated

File: README.md [hunk 32]
```diff
@@ -46,27 +114,71 @@ Defining them is part of the set up of your project.
...
+
+After having all your environment variables, to build a **Domain Linkage Credential**:
+
+- run: `yarn well-known-config`.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md:162]: Thats exactly the same, but ok.

File: README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: rephrase it to: 
`The process is full of interactions between the client (browser), the extension/wallet and the server (backend).`

File: README.md (hunk 13 above)
Comment (kilted-andres) [reply README.md]: Rephrase it to:
`The process is full of interactions between the client (browser), the extension/wallet and the server (backend).`

Comment (weichweich) [reply README.md]: The problem is `full of`. This sounds like it's full. There is no space for another request or your computer would break.

File: README.md (hunk 14 above)
Comment (weichweich) [reply README.md:183]: You would rather write something like `This will help you to understand`.

File: README.md [hunk 33]
```diff
@@ -46,27 +114,71 @@ Defining them is part of the set up of your project.
...
+
+If you want know more about this, check out the [Identity Foundation Documentation](https://identity.foundation/.well-known/resources/did-configuration/).
+[Our dApp documentation also have a section about this.](https://docs.kilt.io/docs/develop/dApp/well-known-did-config)
+
+## Starting the dApp locally
+
+After having all your environment variables and your well-known-did-configuration:
+
+- run `yarn start`.
```
[the remark this answers is not in the document]
Comment (kilted-andres) [reply README.md]: done

File: README.md (hunk 13 above)
Comment (weichweich) [reply README.md]: What do you want to tell the reader with that?

Comment (weichweich) [reply README.md]: The process consists of multiple extension-api messages that are transmitted via HTTP.

Try to use a neutral phrases. full of, abundant are all rather emotional terms. We don't want to convey emotions. Especially not the ones that don't contribute to a nice experience for the reader.

Comment (kilted-andres) [reply README.md]: The other critic was referring to the "S" on HTTP**S**, right?

Comment (kilted-andres) [reply README.md]: `abundant` and `full` are positive terms for me. 

But I can change it to multiple, if it makes you happier.

File: README.md (hunk 14 above)
Comment (kilted-andres) [reply README.md]: It's `This should help you to understand` now.

Comment (kilted-andres) [reply README.md]: It's `This should help you understand` now.**   (grammar)

**Review by weichweich — APPROVED** [review]
LGTM

File: README.md (hunk 11 above)
Comment (Ad96el) [reply README.md:173]: The login process with the diagram. Build is fine and should be here

File: README.md (hunk 12 above)
Comment (Ad96el) [reply README.md:17]: I believe the current semantics dictate that users have to set up an account and password. So one negation has to be removed.

File: .env.example [hunk 34]
```diff
@@ -0,0 +1,8 @@
+WSS_ADDRESS=
+FRONTEND_PORT=
+BACKEND_PORT=
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply .env.example]: You can list your open ports with: 

`sudo lsof -i -P -n | grep LISTEN`

and then kill the process.

File: README.md (hunk 12 above)
Comment (kilted-andres) [reply README.md:17]: Oh, I did not notice. Thanks!

The sentence is gone now though.
